app/models/user.rb

```diff
@@ -1,2 +1,15 @@
 class User < ApplicationRecord
+  has_secure_password
+
+  has_many :folders
+  has_many :uploads, through: :folders
+
+  validates :username,        presence: true, uniqueness: true
+  validates :cellphone,       presence: true, uniqueness: true
+  validates :email,           presence: true, uniqueness: true
+  validates :password_digest, presence: true
+
+  def root_folder
```

Comment:
The 'username' column for 'users' table has been removed and replaced by 'first_name' and 'last_name'.